discovery/rendezvous.go

```diff
@@ -0,0 +1,159 @@
+package discovery
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/rand"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/status-im/rendezvous"
+)
+
+const (
+	registrationPeriod = 10 * time.Second
+	bucketSize         = 10
+)
+
+func NewRendezvous(servers []ma.Multiaddr, identity *ecdsa.PrivateKey, node *discover.Node) (*Rendezvous, error) {
+	r := new(Rendezvous)
+	r.servers = servers
+	r.registrationPeriod = registrationPeriod
+	r.bucketSize = bucketSize
+
+	r.record = enr.Record{}
+	r.record.Set(enr.IP(node.IP))
+	r.record.Set(enr.TCP(node.TCP))
+	r.record.Set(enr.UDP(node.UDP))
+	if err := enr.SignV4(&r.record, identity); err != nil {
+		return nil, err
+	}
+	return r, nil
+}
+
+// Rendezvous is an implementation of discovery interface that uses
+// rendezvous client.
+type Rendezvous struct {
+	mu     sync.RWMutex
+	client *rendezvous.Client
+
+	servers            []ma.Multiaddr
+	registrationPeriod time.Duration
+	bucketSize         int
+	record             enr.Record
+}
+
+func (r *Rendezvous) Running() bool {
+	r.mu.RLock()
+	defer r.mu.RUnlock()
+	return r.client != nil
+}
+
+// Start creates client with temporary (not persisted) identity.
```

Comment:
What is `temporary identity`? Is identity of rendezvous something different than node identity (which is a key pair).

Reply:
i meant that key is generated in the client itself, and identity will be discarded once client is destroyed

Reply:
So it's a client identity? Not really understand the purpose of the identity here, what it is used for and how it is different from the `p2p.Server` identity.

Reply:
yes, it is client identity. it is used to establish secure communication. similar to ethereum. see:

https://github.com/libp2p/go-libp2p-secio/blob/master/protocol.go#L137

you are right that i can use the key from p2p.Server. but why it would be preferable?

Reply:
Ephemeral sounds better. i will make an update in rendezvous later